xapian-letor/api/featurevector_internal.cc

```diff
@@ -117,3 +117,31 @@ FeatureVector::Internal::get_nonzero_num(){
     //cout << "nonzero: "<< nonzero <<endl;
     return nonzero;
 }
+
+void
+FeatureVector::Internal::normalise() {
```

Comment:
Looking at this a little further, I'm not convinced this function does what is needed. Is it normalising the feature vector, so its length is 1, or should it be normalising each feature's values across the range of feature vectors. Because this code seems to do neither. Perhaps some tests would help clarify this?